app/src/main/java/com/example/yard/Post.java

```diff
@@ -1,14 +1,145 @@
 package com.example.yard;
 
+import androidx.annotation.Nullable;
+import androidx.annotation.RequiresApi;
 import androidx.appcompat.app.AppCompatActivity;
-
+import androidx.core.content.FileProvider;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.net.Uri;
+import android.os.Build;
 import android.os.Bundle;
+import android.os.Environment;
+import android.provider.MediaStore;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.parse.ParseException;
+import com.parse.ParseFile;
+import com.parse.ParseUser;
+import com.parse.SaveCallback;
+
+import java.io.File;
 
+@RequiresApi(api = Build.VERSION_CODES.O)
 public class Post extends AppCompatActivity {
 
+    public static final int CAPTURE_IMAGE_ACTIVITY_REQUEST_CODE = 45;
+    public static final String TAG = "Post";
+    private Button takePicture;
+    private Button submit;
+    private TextView etdescription;
+    private Button upload;
+    private File photoFile;
+    private ImageView image;
+    private String photoFileName = "photo.jpg";
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_post);
+
+        takePicture = findViewById(R.id.btnTakePicture);
+        submit = findViewById(R.id.btnSubmit);
+        upload = findViewById(R.id.btnUploadPicture);
+        etdescription = findViewById(R.id.tvDescription);
+        image = findViewById(R.id.ivImage);
+        takePicture.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                launchcamera();
+            }
+        });
+        submit.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                String description = etdescription.getText().toString();
+
+                if (description.isEmpty()){
+                    Toast.makeText(Post.this, "Description cannot be empty", Toast.LENGTH_SHORT).show();
+                    return;
+                }
+//                if (image.getDrawable() == null){
+//                    Toast.makeText(Post.this, "No picture Uploaded", Toast.LENGTH_SHORT).show();
+//                }
+                ParseUser currentUser = ParseUser.getCurrentUser();
+                savePost(description, currentUser, photoFile);
+            }
+        });
+    }
+
+    private void launchcamera() {
+        Intent intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+        // Create a File reference for future access
+        photoFile = getPhotoFileUri(photoFileName);
+        // wrap File object into a content provider
+        // required for API >= 24
```

Comment:
remove the comments copied from guidance. They are here for you to understand it before using the codes. Seem for the rest of the commends copied from the guidance.